book/src/assets/how_to_use_assets.md

```diff
@@ -0,0 +1,221 @@
+
+* amethyst_assets
+* amethyst_audio
+* amethyst_gltf
+* amethyst_locale
+* amethyst_ui
+
+## Steps
+
+1. Instantiate the Amethyst application with the assets directory.
+
+    ```rust,no_run,noplaypen
+    # extern crate amethyst;
+    #
+    use amethyst::{
+        prelude::*,
+        utils::application_root_dir,
+    };
+    #
+    # pub struct LoadingState;
+    # impl SimpleState for LoadingState {}
+
+    fn main() -> amethyst::Result<()> {
+        // Sets up the application to read assets in
+        // `<app_dir>/assets`
+        let app_root = application_root_dir()?;
+        let assets_dir = app_root.join("assets");
+
+        //..
+    #   let game_data = GameDataBuilder::default();
+
+        let mut game = Application::new(assets_dir, LoadingState, game_data)?;
+    #
+    #   game.run();
+    #   Ok(())
+    }
+    ```
+
+2. Ensure that the [`Processor<A>` system][doc_processor_system] for asset type `A` is registered in the dispatcher.
+
+    For asset type `A`, `Processor<A>` is a `System` that will asynchronously load `A` assets. Usually the crate that provides `A` will also register `Processor<A>` through a `SystemBundle`. Examples:
+
+    * `FontAsset` is provided by `amethyst_ui`, `UiBundle` registers `Processor<FontAsset>`.
+    * `Source` is provided by `amethyst_audio`, `AudioBundle` registers `Processor<Source>`.
+    * `SpriteSheet` is provided by `amethyst_renderer`, `RenderBundle` registers `Processor<SpriteSheet>` given `.with_sprite_sheet_processor()` is called before adding it to the `GameDataBuilder`.
+
+    ```rust,no_run,noplaypen
+    # extern crate amethyst;
+    #
+    # use amethyst::{
+    #     prelude::*,
+    #     renderer::{DisplayConfig, Pipeline, RenderBundle, Stage},
+    #     ui::{DrawUi, UiBundle},
+    #     utils::application_root_dir,
+    # };
+    #
+    # pub struct LoadingState;
+    # impl SimpleState for LoadingState {}
+
+    fn main() -> amethyst::Result<()> {
+    #   let app_root = application_root_dir()?;
+    #   let assets_dir = app_root.join("assets");
+    #
+    #   let display_config = DisplayConfig::default();
+    #   let pipeline = Pipeline::build().with_stage(
+    #       Stage::with_backbuffer()
+    #           .clear_target([0.; 4], 0.)
+    #           .with_pass(DrawUi::new()),
+    #   );
+        // ..
+
+        let game_data = GameDataBuilder::default()
+            .with_bundle(UiBundle::<String, String>::new())?
+            .with_bundle(
+                RenderBundle::new(pipeline, Some(display_config))
+                    .with_sprite_sheet_processor()
+            )?;
+
+        // ..
+    #   let mut game = Application::new(assets_dir, LoadingState, game_data)?;
+    #
+    #   game.run();
+    #   Ok(())
+    }
+    ```
+
+3. Use the [`Loader`][doc_loader] resource to load the asset.
+
+    ```rust,no_run,noplaypen
+    # extern crate amethyst;
+    use amethyst::{
```

Comment:
To be consistent with the rest of the examples here, this should probably be commented out while imo `LoadingState` would be helpful to show to the reader, although at what point that should be done I'm not sure, since the type is introduced in Step 1.

Reply:
Oh yes, shall comment the imports and uncomment `LoadingState` here, since it's the focus of this step